cmd/yggctl/actions.go

```diff
@@ -209,3 +188,23 @@ func generateMessage(messageType, responseTo, directive, content string, metadat
 		return nil, fmt.Errorf("unsupported message type: %v", messageType)
 	}
 }
+
+func connectBus() (*dbus.Conn, error) {
+	var connect func(...dbus.ConnOption) (*dbus.Conn, error)
+	var conn *dbus.Conn
+	var err error
+	if os.Getenv("DBUS_SESSION_BUS_ADDRESS") != "" && os.Geteuid() > 0 {
+		connect = dbus.ConnectSessionBus
+	} else {
+		connect = dbus.ConnectSystemBus
+	}
+
+	conn, err = connect()
+	if err != nil {
+		if os.Getenv("DBUS_SESSION_BUS_ADDRESS") != "" {
```

Comment:
This line should be changed to:
```go
if os.Getenv("DBUS_SESSION_BUS_ADDRESS") != "" && os.Geteuid() > 0 {
```
otherwise this function will return incorrect error messages for the case you are trying to solve.

Reply:
But I have better proposal to avoid checking conditions twice. What about this:
```go
func connectBus() (*dbus.Conn, error) {
	var connect func(...dbus.ConnOption) (*dbus.Conn, error)
	var conn *dbus.Conn
	var err error
	var errMsg string
	if os.Getenv("DBUS_SESSION_BUS_ADDRESS") != "" && os.Geteuid() > 0 {
		connect = dbus.ConnectSessionBus
		errMsg = "cannot connect to session bus (" + os.Getenv("DBUS_SESSION_BUS_ADDRESS") + "): %w"
	} else {
		connect = dbus.ConnectSystemBus
		errMsg = "cannot connect to system bus: %w"
	}

	conn, err = connect()
	if err != nil {
		return nil, fmt.Errorf(errMsg, err)
	}
	return conn, nil
}
```

Reply:
Nice. Yea. This should work. And I completely forgot my Geteuid check when I rewrote the function. :facepalm: